### PR TITLE
Add Typescript definitions

### LINKED ISSUE #2: Add typings for imports in Typescript
Getting a `Could not find a declaration file for module 'human-signals'.` when importing this in Typescript, such as in the following:

```typescript
import {signalsByName} from 'human-signals';
```

Also `@types/human-signals` does not exist in npm.

Happy to assist if needed.

### PR DESCRIPTION
**Pull request by electrovir**
This resolves #2

Simple solution which is mostly just a copy paste straight from the README. Comments included.

- [x] I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have added tests (we are enforcing 100% test coverage).
  - **Not sure how to add tests for this.**
- [ ] I have added documentation in the `README.md`, the `docs` directory (if
      any) and the `examples` directory (if any).
  - **Also not sure exactly what to document, if anything, in the README about this.**
- [x] The status checks are successful (continuous integration). Those can be
      seen below.

### CODE REVIEW

**Comment by electrovir** [comment]
Should I update `.all-contributorsrc`?

File: src/main.d.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  | 'ansi'
+  | 'posix'
+  | 'bsd'
+  | 'systemv'
+  | 'other'
+
+export declare type Signal = {
+  /**
+   * Standard name of the signal, for example 'SIGINT'.
+   */
```
Comment (electrovir) [inline src/main.d.ts]: Comments above each property so they are nicely displayed in code editor auto completions.

**Comment by electrovir** [comment]
Example type tips rendered in VS Code:

Screen 2020-03-05 at 8 01 44 PM
Screen 2020-03-05 at 8 02 18 PM

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "human-signals",
-  "version": "2.0.0",
+  "version": "2.1.0",
```
Comment (ehmicky) [inline package.json]: There is a Gulp task [`gulp releaseMinor`](https://github.com/ehmicky/dev-tasks#gulp-releasepatch-gulp-releaseminor-gulp-releasemajor) that increments package version, create a GitHub release, waits for CI to succeed then publishes to npm. This is done on `master`, i.e. outside of PRs.
Therefore, would it be possible to remove the `version` change (also in `package-lock.json`)?

File: src/main.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  name: string
+  /**
+   * Code number of the signal, for example 2. While most number are cross-platform, some are different between different OS.
+   */
+  number: number
+  /**
+   * Human-friendly description for the signal, for example 'User interruption with CTRL-C'.
+   */
+  /** this is the description description */
```
Comment (ehmicky) [inline src/main.d.ts]: This comment `this is the description description` should be removed?

File: src/main.d.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Object whose keys are signal names and values are signal objects.
+ */
+export declare const signalsByName: { [signalName: string]: Signal }
+/**
+ * Object whose keys are signal numbers and values are signal objects.
+ */
+export declare const signalsByNumber: { [signalNumber: number]: Signal }
```
Comment (ehmicky) [inline src/main.d.ts]: `signalNumber` is actually a `string`: `{ '50': ... }`.

**Comment by ehmicky** [comment]
@all-contributors could you please add @electrovir for adding this code?

File: package.json (hunk 2 above)
Comment (electrovir) [reply package.json]: Fixed. Should I leave my update to `CHANGELOG.md`?

File: src/main.d.ts (hunk 3 above)
Comment (electrovir) [reply src/main.d.ts]: Oo nice catch!

File: src/main.d.ts (hunk 4 above)
Comment (electrovir) [reply src/main.d.ts]: Fixed

File: package.json (hunk 2 above)
Comment (ehmicky) [reply package.json]: Yes it's good :+1:. Thanks!

**Pull request merged**